### ISSUE TITLE
Inconsistent object mapping behavior when querying using LINQ  vs `QueryToListAsync`

### ISSUE THREAD

**Issue by jeffijoe** [linq2db/linq2db#4437]
### Describe your issue

When using `QueryToListAsync<Test>` where `Test` is a record **with a primary constructor** and `[property: Column("some_column")]` attributes, it appears the column names specified in the attributes are not being used. If the record does not have a constructor, then it works fine. Additionally, using `connection.GetTable<MyType>()` to query, then the record with a primary constructor also works. Removing the constructor makes it work with `QueryToListAsync`.

### Steps to reproduce

Using the following as our table/query result type

```csharp
[Table("test")]
public record Test(
    [property: Column("some_column")]
    string SomeColumn);
```

When using the following:

```csharp
await connection.QueryToListAsync<Test>("select some_column from test")
```

The resulting records will have `null` in `SomeColumn` because the `[Column]` attribute which provides the correct name is not being used.

If instead I do

```csharp
await db.GetTable<Test>().FirstOrDefaultAsync();
```

Then the column mapping works and I get the expected value in `SomeColumn`.

If I change my record to not have a constructor, then everything works fine with `QueryToListAsync` as well:

```csharp
[Table("test")]
public record Test
{
    [Column("some_column")]
    public string SomeColumn { get; init; }
}
```

Is there a difference in how Linq2Db handles object mapping for LINQ-based queries vs SQL-based ones?

### Environment details

`Linq To DB` version: 5.4

Database (with version): Postgres 13

ADO.NET Provider (with version): Npgsql 8.0

Operating system: macOS

.NET Version: 8.0

**Comment by MaceWindu** [comment]
Yes, currently mapping for raw queries works in a different way. We will unify it in upcoming 6.0 release

**Comment by jeffijoe** [comment]
Thank you @MaceWindu, and thank you for all the work you guys are doing on Linq2Db, it's an incredible library! 🙏

**Comment by MaceWindu** [comment]
We reviewed this issue and currently it is expected behavior. We map raw SQL by name in C#, so proper query should be 

```cs
// note alias added for column
await connection.QueryToListAsync<Test>("select some_column AS ColumnName from test")
```

Changing this will be:
- breaking change to existing users
- will break multi-table queries with same column name (`SELECT table1.field, table2.field FROM ...`)

We don't like current situation, but changes to it should be carefully designed

**Comment by jeffijoe** [comment]
That makes sense, thank you! Perhaps adding something to the documentation about it would help people encountering the behavior and expecting the same thing as I was?
